packages/flutter/lib/src/material/text_field.dart

```diff
@@ -663,8 +675,11 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
         inputFormatters: formatters,
         rendererIgnoresPointer: true,
         cursorWidth: widget.cursorWidth,
-        cursorRadius: widget.cursorRadius,
-        cursorColor: widget.cursorColor ?? themeData.cursorColor,
+        cursorRadius: _cursorRadius,
+        cursorColor: widget.cursorColor ?? Theme.of(context).cursorColor,
```

Comment:
Feel free to file a bug for later but the Material Theme default cursor color on iOS can arguably be the stock iOS color